source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -1508,6 +1508,40 @@ bool SwiftLanguageRuntime::GetDynamicTypeAndAddress_Class(
   if (error.Fail())
     return false;
 
+  // If LLDB succeeded on its own, check whether RemoteAST succeeded.
+  if (class_type_or_name.GetCompilerType().IsValid()) {
+    swift::remoteAST::RemoteASTContext remote_ast(
+                                         *swift_ast_ctx->GetASTContext(),
+                                         GetMemoryReader());
+
+    swift::Type expected_type =
+      GetSwiftType(class_type_or_name.GetCompilerType());
+    swift::remote::RemoteAddress instance_address(in_value.GetPointerValue());
+    auto metadata_address =
+      remote_ast.getHeapMetadataForObject(instance_address);
+    if (!metadata_address) {
+      llvm::errs() << "Remote AST failed to get metadata address for type "
+                   << expected_type.getString() << "\n";
+      llvm_unreachable(
```

Comment:
Don't use llvm_unreachable here.  This operation, could fail, for instance, because the stack or heap was corrupted, and you certainly wouldn't want lldb aborting in that instance.  Finding dynamic types is an operation it's okay to fail at.  Just emit something useful to the types log and return false.

Reply:
Sorry it wasn't clear from my commit message/Do Not Merge designation; I wanted a hard failure here because it would indicate a case where LLDB knew how to resolve something that Remote AST couldn't, which would be a bug we need to fix in RemoteAST.